### ISSUE TITLE
Allow `defaultProps` for pure functional React components

### ISSUE THREAD

**Issue by frostney** [planttheidea/moize#19]
When using a pure function React component with `defaultProps`, `moize` will ignore them.

```jsx
const HelloComponent = ({ name }) => <div>Hello {name}</div>;

HelloComponent.defaultProps = {
  name: 'moize'
};
```

On the other hand, passing it as a default parameter value works:
```jsx
const HelloComponent = ({ name = 'moize' }) => <div>Hello {name}</div>;
```

**Comment by planttheidea** [comment]
Ah interesting ... this is probably true of all static properties (`contextTypes` and `propTypes` as well). I'll be releasing a fix for this later tonight (already written, just need to add a test for it), and I'll close then.

**Comment by planttheidea** [comment]
Published under `2.0.1`, closing.
